### PR TITLE
fix(worker, application-generic): Remove In-App message lookup for stateless Workflows

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?

* Remove In-App message lookup for stateless Workflows
  * In-app messages support edit after delivery by modifying the message entity if it already exists in the database. Stateless Workflows do not persist Workflow (formerly Template) and Step (formerly NotificationTemplate) entities and therefore have undefined identifiers. During in-app message delivery, Stateless Workflows with multiple in-app steps were retrieving the Message entity of the previous in-app step in the same Workflow due to the Workflow and Step identifiers being undefined. This retrieved Message entity meant the logic to create a new message was never executed, the effect being that Stateless Workflow executions with multiple in-app steps only ever had 1 in-app message delivered.
  * This issue didn't surface at compile because the database repository typings are broken, allowing for `any | undefined` attributes to be passed in
* Add tests for double in-app Workflows

### Screenshots

_TDD (before)_
```bash
    1) should store 2 in-app messages for a single notification event [stateless]


  1 passing (3s)
  1 failing

  1) Bridge Trigger
       should store 2 in-app messages for a single notification event [stateless]:

      AssertionError: expected 1 to equal 2
      + expected - actual

      -1
      +2
      
      at /Users/rifont/git/novu/apps/api/src/app/events/e2e/bridge-trigger.e2e.ts:707:41
      at Generator.next (<anonymous>)
      at fulfilled (src/app/events/e2e/bridge-trigger.e2e.ts:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

_TDD (after)_
```bash
    ✔ should store 2 in-app messages for a single notification event [stateless] (149ms)


  2 passing (2s)
```

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/worker/src/app/workflow/usecases/send-message/send-message.command.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export class SendMessageCommand extends EnvironmentWithUserCommand {
   notificationId: string;
 
   @IsOptional()
-  _templateId: string;
+  _templateId?: string;
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/send-message/send-message.command.ts:31]: This was missing and causing erroneous type usage, see above that the same attribute is marked via class-validators as being optional.

File: libs/application-generic/src/usecases/execute-bridge-request/execute-bridge-request.usecase.ts [hunk 2]
```diff
@@ -120,7 +120,9 @@ export class ExecuteBridgeRequest {
         } else {
           // Handle unknown bridge request errors
           Logger.error(
-            `Unknown bridge request error calling \`${url}\`: \`${body}\``,
+            `Unknown bridge request error calling \`${url}\`: \`${JSON.stringify(
+              body
+            )}\``,
```
Comment (rifont) [inline libs/application-generic/src/usecases/execute-bridge-request/execute-bridge-request.usecase.ts:125]: The `body` object was not serializing to logs correctly (previously `[Object object]`), this fixes that.

File: apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts [hunk 3]
```diff
@@ -179,31 +200,34 @@ export class SendMessageInApp extends SendMessageBase {
...
     const inAppMessage = inAppMessageFromBridgeOutputs(bridgeOutputs);
 
+    const channelData: Partial<Pick<MessageEntity, 'content' | 'subject' | 'avatar' | 'payload' | 'cta' | 'tags'>> = {
+      content: (this.storeContent() ? inAppMessage.content || content : null) as string,
+      cta: bridgeOutputs ? inAppMessage.cta : step.template.cta,
+      subject: inAppMessage.subject,
+      avatar: inAppMessage.avatar,
+      payload: messagePayload,
+      tags: command.tags,
+    };
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts:211]: Constructing the channel data for shared use in both message creation and update logic.

File: apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts [hunk 4]
```diff
@@ -213,10 +237,8 @@ export class SendMessageInApp extends SendMessageBase {
         {
           $set: {
             seen: false,
-            cta: step.template.cta,
-            content,
-            payload: messagePayload,
             createdAt: new Date(),
+            ...channelData,
```
Comment (rifont) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts:243]: Ensuring that the full constructed channel data is passed to the Message update logic too.

File: apps/api/src/app/events/e2e/bridge-trigger.e2e.ts [hunk 5]
```diff
@@ -680,6 +680,35 @@ contexts.forEach((context: Context) => {
...
+        _subscriberId: subscriber._id,
+        templateIdentifier: workflowId,
+        channel: StepTypeEnum.IN_APP,
+      });
+
+      expect(sentMessages.length).to.be.eq(2);
+      const messageBodies = sentMessages.map((message) => message.content);
+      expect(messageBodies).to.include('Hello there 1');
+      expect(messageBodies).to.include('Hello there 2');
+    });
```
Comment (rifont) [inline apps/api/src/app/events/e2e/bridge-trigger.e2e.ts:711]: This newly added test was previously failing for Stateless Workflows (per PR description).

**Review by davidsoderberg — APPROVED** [review]

**Pull request merged**